### ISSUE TITLE
[Idea] `visible` assertion considers color contrasts

### ISSUE THREAD

**Issue by jorgelainfiesta** [thefrontside/interactors#15]
## Motivation

The premise of Interactos is to enable developers to test their apps as a user would use them. Thus, the question of something being "visible" becomes more sophisticated as not all users are the same.  In a later iteration of Interactors, we can take accessibility one step forward and bake color contrast in the visible scheme for filters.

For instance, if an assertion says there should be a visible button with text "Submit", and it is there on the DOM but the color contrast is not enough then BIgTest would fail the assertion.

I think this feature would give a powerful argument to a unique selling point that we have: accessibility at the core. I can imagine landing pages, talks, and articles about Interactors once we make this narrative more robust.

It is also in line with our Leitmotif: work less, test more.

## Approach

I'm not sure how we can assess color contrast, but it must be done over the rendered element to account for background-images. It wouldn't be enough to compare `background` vs `color`. I've experienced false positives with tools that check color contrasts with that approach.

This could be an opt-in setup, and we can allow the user to select how restrictive they want their assertions to be. For example they could enable only `AAA` contrast, or be ok with `A` contrast. We can even issue a little warning when they set the contrast setting too low.
